antiphishing/antiphishing.py

```
@@ -1,357 +1,895 @@
+                log.warning(f"HTTP error following redirects for {url} at step {current_url}: {e.status} {e.message}")
+                # Add the hostname even if it errored, it might still be malicious
+                try:
+                    parsed = urlsplit(current_url)
+                    if parsed.netloc:
+                        redirect_chain.append(parsed.netloc.lower())
+                except ValueError: pass
+                break # Stop following on client/server errors
+            except aiohttp.ClientError as e:
+                log.warning(f"Network error following redirects for {url} at step {current_url}: {e}")
+                break # Stop following on network errors
+            except asyncio.TimeoutError:
```

Ruff (0.8.2)

Undefined name `asyncio`

(F821)